src/GraphQL/DI/IGraphQLBuilder.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+
+namespace GraphQL.DI
+{
+    /// <summary>
+    /// An interface for configuring GraphQL.NET services.
+    /// </summary>
+    public interface IGraphQLBuilder
```

Comment:
Looks more like `IServiceRegister` https://github.com/EasyNetQ/EasyNetQ/blob/develop/Source/EasyNetQ/DI/IServiceRegister.cs

Reply:
Could be. I’ve only ever used MS DI, and occasionally AutoFac

Reply:
We should discuss in a separate issue or thread; it will get lost here.

Having IGraphQLBuilder simply point to IServiceRegister does not seem to change anything -- then IServiceRegister will have all of the Register and TryRegister methods.  What would be the point of the builder interface?

Are you just refactoring so the effect to the user is that `.AddDataLoader()` is not mixed with `.Register` and `.TryRegister`?  That might make sense, I think...  Of course it really only would eliminate 3 methods (Register, TryRegister, and Configure) and adds 1 (ServiceRegister).

Reply:
> Are you just refactoring so the effect to the user is that .AddDataLoader() is not mixed with .Register and .TryRegister?

Yes. `IGraphQLBuilder` != `IServiceRegister`. Now they actually **are** equal techically but semantical they are different.

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/issues/2771